python/lsst/meas/base/afterburner.py

```diff
@@ -0,0 +1,183 @@
+
+class AfterburnerConfig(lsst.pex.config.Config):
+    '''
+    Default AfterburnerConfig. Currently this is an empty list, meaning that there are no default plugins run.
+    The config object for each plugin must use this variable to specify the names of all plugins to be run.
+    '''
+    plugins = AfterburnerPlugin.registry.makeField(multi=True, default=[], doc="Plugins to be run and their "
+                                                    "configuration")
+    pass
+
+class AfterburnerTask(lsst.pipe.base.Task):
+    '''
+    This task facilitates running plugins which will operate on a source catalog. These plugins may do things
+    such as classifying an object based on source record entries inserted during a measurement task.
+
+    Plugins may either take an entire catalog to work on at a time, or 
+    '''
+    ConfigClass = AfterburnerConfig
+    _DefaultName = "afterburner"
+
+    def __init__(self, schema, plugMetadata=None, **kwargs):
+        """
+        Constructor; only called by derived classes
+
+        @param[in] plugMetaData     An lsst.daf.base.PropertyList that will be filled with metadata
+                                    about the plugins being run. If None, an empty empty PropertyList
+                                    will be created.
+        @param[in] **kwargs         Additional arguments passed to lsst.pipe.base.Task.__init__.
+        """
+        lsst.pipe.base.Task.__init__(self,**kwargs)
+        self.schema = schema
+        if plugMetadata is None:
+            plugMetadata = lsst.daf.base.PropertyList()
+        self.plugMetadata = plugMetadata
+
+        self.initializePlugins()
+
+    def initializePlugins(self):
+        '''
+        Initialize the plugins according to the configuration.
+        '''
+
+        pluginType = namedtuple('pluginType','single multi')
+        self.executionDict = {}
+        # Read the properties for each plugin. Allocate a dictionary entry for each run level. Verify that
+        # the plugins are above the minimum run level for an afterburner plugin. For each run level, the
+        # plugins are sorted into either single record, or multi record groups to later be run appropriately
+        for executionOrder, name, config, PluginClass in self.config.plugins.apply():
+            if executionOrder not in self.executionDict:
+                self.executionDict[executionOrder] = pluginType(single=[], multi=[])
+            if PluginClass.getExecutionOrder() >= BasePlugin.DEFAULT_AFTERBURNER:
+                plug = PluginClass(config, name, self.schema, metadata=self.plugMetadata)
+                if plug.PlugType == 'single':
+                    self.executionDict[executionOrder].single.append(plug)
+                elif plug.PlugType == 'multi':
+                    self.executionDict[executionOrder].multi.append(plug)
+            else:
+                raise ValueError("{} has an execution order less than the minimum for an afterburner plugin."
```

Comment:
Need a space before the end quote, or the next line will be right up against the period.